src/components/tab/Tab.tsx

```diff
@@ -1,14 +1,22 @@
+import { Dispatch, SetStateAction } from 'react'
 import { useTranslation } from 'react-i18next'
 import Box from '@mui/material/Box'
 
 import useFilter from '~/hooks/table/use-filter'
 
-import { styles } from './Tab.styles'
+import { TabsInfoItem, ExternalFilter, Options } from '~/types/common/types/common.types'
+import { styles } from '~/components/tab/Tab.styles'
 
-const Tab = ({ tab, setTab, activeTab }) => {
+interface TabProps<U> {
+  tab: TabsInfoItem<Options<U>>
+  setTab: Dispatch<SetStateAction<ExternalFilter>>
+  activeTab: string
+}
+
+const Tab = <T, U>({ tab, setTab, activeTab }: TabProps<U>) => {
```

Comment:
It is hard to understand what is T for and what is U for.

Reply:
fixed